api/operator_keyring.go

```diff
@@ -16,6 +16,9 @@ type KeyringResponse struct {
 	// Segment has the network segment this request corresponds to.
 	Segment string
 
+	// Messages has information or errors from serf
+	Messages map[string]string `json:",omitempty"`
```

Comment:
Does it matter that these messages will be randomly ordered? If you stored them as something like `[][]string` instead of `map[string]string` then at least you could retain the ordering.

Reply:
`Messages` are already in the response from https://github.com/hashicorp/consul/blob/884b2e00af8bdde14ca5d0f79cde77929e35664b/agent/structs/structs.go#L1337-L1345
I only added it here so that it gets deserialized. 

Since the results are collected in parallel from all the nodes, I wouldn't even be sure how to order them. The order of arrival could be different every time as well.